time/timesyncd.go

```diff
@@ -0,0 +1,175 @@
+package time
```

Comment:
To avoid clash with the go time package, I'd go for something else, like `timesync`.

```suggestion
package timesync
```

Reply:
I wanted to have the same prefix for the DBus interfaces controlling other time-related features in the future, and so far we've got it synchronized with package naming. In the end the only clash is in the main file, where it can be simply aliased.

Reply:
To me this is mostly about filling the gap of missing D-Bus interfaces in systemd. Ideally, we would extend systemd, but this is much more time consuming, so I am fine doing our thing now here.

But, with that in mind, I think it make sense to align with the systemd D-Bus naming: Systemd has two time/date related D-Bus services: `org.freedesktop.timesync1` and `org.freedesktop.timedate1`.  What we currently miss is setting the system NTP server, which most likely should be part of the `org.freedesktop.timesync1` service.

Reply:
Rather than the Systemd D-Bus naming, consider naming of our D-Bus interfaces - now it's `io.hass.os.Time.Timesyncd`, which is implemented in `time/timesyncd.go`. If we added e.g. interface for timezone setting (leave aside the fact there's some method in `org.freedesktop.timedate1` that _might_ be used instead), a logical path would be `io.hass.os.Time.Timezone` and the implementation would go to `time/timezone.go`. If the package should be renamed, then we should also adjust D-Bus paths and interfaces.

Reply:
With the other discussion in mind, it kinda seems what we need to build here is mostly a systemd config file manager. So aligning the naming to the config file names probably make sense :thinking: